### PR TITLE
#18 base building

### PR DESCRIPTION
**Pull request by jetenergy**

### CODE REVIEW

File: Hivemind/Assets/Scripts/BaseBuilding/BaseTile.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        if (CurrTile != null || IsIndestructable) return;
+
+        CurrTile = Instantiate(gObj);
+        CurrTile.transform.SetParent(gameObject.transform, false);
+        CurrTile.transform.localRotation = Quaternion.Euler(0, (DefaultRotation < 0) ? Random.Range(0, 5) * 60 : DefaultRotation, 0);
+
+        RoomScript = CurrTile.GetComponent<BaseRoom>();
+    }
+
+    internal void DestroyRoom()
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/BaseBuilding/BaseTile.cs]: Recommend deleting this method and defaulting bool forced to false in the method below

File: Hivemind/Assets/Scripts/BaseBuilding/Rooms/BaseRoom.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/BaseBuilding/Rooms/BaseRoom.cs]: Remove unused usings pls

File: Hivemind/Assets/Scripts/BaseBuilding/Rooms/BaseUnitRoom.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        if (unitGroup.AddUnit())
+        {
+            if (UnitResource != null)
+            {
+                GameObject ant = Instantiate(Resources.Load(UnitResource) as GameObject);
+
+                GameObject container = GameObject.Find("Ants");
+                if (container == null)
+                {
+                    throw new Exception("The \"Ants\" container was not found. Please add a container to save the Hiearchy from clutter!");
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/BaseBuilding/Rooms/BaseUnitRoom.cs]: Hierarchy not Hiearchy

File: Hivemind/Assets/Scripts/BaseBuilding/UnitRoomPriorityQueue.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+class UnitRoomPriorityQueue
+{
+    List<BaseUnitRoom> queue = new List<BaseUnitRoom>();
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/BaseBuilding/UnitRoomPriorityQueue.cs]: queue should be private

File: Hivemind/Assets/Scripts/Ui/UiController.cs [hunk 5]
```diff
@@ -20,15 +20,13 @@ public class UiController : MonoBehaviour, IInitializePotentialDragHandler, IDra
 
     [SerializeField] private TextMeshProUGUI resourceTextBox;
 
-    [SerializeField] private UnitController unitController;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/Ui/UiController.cs:23]: Remove autors comment block above pls (can't comment there, not relevant to this line)

File: Hivemind/Assets/Scripts/UnitController.cs [hunk 6]
```diff
@@ -42,8 +45,83 @@ public void SetMaxUnits(Guid unitGroupId, int amount)
...
+            if (child.GetComponent<Ant>() != null)
+            {
+                ants.Add(child.GetComponent<Ant>());
+            }
+            else
+            {
+                throw new Exception("The ant doesnt have the required script!");
+            }
+        }
+
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/UnitController.cs:77]: Empty line

File: Hivemind/Assets/Scripts/UnitGroup.cs [hunk 7]
```diff
@@ -29,28 +29,59 @@ internal UnitGroup(GameObject unitIconBase)
         UpdateText();
     }
 
-    public void SetMaxUnits(int amount)
+    public bool RemoveMax()
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/UnitGroup.cs]: Change to void pls, same goes for methods below if they are not used as bools

File: Hivemind/Assets/Scripts/UnitGroup.cs [hunk 8]
```diff
@@ -29,28 +29,59 @@ internal UnitGroup(GameObject unitIconBase)
...
     private void UpdateText()
     {
-        textBox.text = $"{CurrentUnits}/{MaxUnits}";
+        if (textBox != null)
+            textBox.text = $"{CurrentUnits}/{MaxUnits}";
+    }
+
+    internal void MergeGroupIntoThis(UnitGroup other)
+    {
+        this.MaxUnits += other.MaxUnits;
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/UnitGroup.cs]: this. is not useful pls remove

File: Hivemind/Assets/Scripts/UnitGroupList.cs [hunk 9]
```diff
@@ -83,15 +104,23 @@ internal void MoveUnit(UnitGroup unit, GameObject groupGameObject)
...
     }
 
-    public MindGroup GetMindGroupFromIndex(int Index)
+    public void DeleteUnitGroup(UnitGroup unitGroup)
+    {
+        GetMindGroupFromUnitId(unitGroup.UnitGroupId)?.RemoveUnit(unitGroup);
+
+        UnityEngine.Object.Destroy(unitGroup.Ui_IconObj);
+    }
+ public MindGroup GetMindGroupFromIndex(int Index)
```
Comment (ErikLouwerse) [inline Hivemind/Assets/Scripts/UnitGroupList.cs]: spacing

**Review by ErikLouwerse — CHANGES_REQUESTED** [review]

File: Hivemind/Assets/Scripts/Ui/UiController.cs (hunk 5 above)
Comment (ErikLouwerse) [reply Hivemind/Assets/Scripts/Ui/UiController.cs:23]: Same goes for al other author comment in the solution

**Review by MMCQ33 — CHANGES_REQUESTED** [review]
fix erik comments

**Review by ErikLouwerse — APPROVED** [review]

**Review by MMCQ33 — APPROVED** [review]

**Pull request merged**